crates/cheatcodes/src/inspector.rs

```diff
@@ -200,6 +200,174 @@ pub struct Cheatcodes {
     /// Breakpoints supplied by the `breakpoint` cheatcode.
     /// `char -> (address, pc)`
     pub breakpoints: Breakpoints,
+
+    /// Track if cool cheatcode was called on each address
+    pub addresses: rHashMap<Address, AddressState>,
+    /// Track if an addresses storage slot is cool or not
+    pub address_storage: rHashMap<Address, rHashMap<U256, StorageSlotState>>,
+    /// How much gas to charge in the next step (op code) based on cool cheatcode calculations
+    pub additional_gas_next_op: u64,
+}
+
+/// Whether an Address is accessed or not
+#[derive(Clone, PartialEq, Debug)]
+pub enum AddressState {
+    /// if already accessed, then charge WARM_STORAGE_READ_COST (100)
+    Warm,
+    /// charge COLD_ACCOUNT_ACCESS_COST (2600)
+    Cool,
+}
+
+/// Whether a Storage Slot is warm or already been modified
+#[derive(Clone, PartialEq, Debug)]
+pub enum StorageSlotState {
+    /// charge extra based on SSTORE calculations
+    WarmWithSLOAD,
+    /// if SSTORE already happened, don't charge extra
+    WarmWithSSTORE,
+    /// same as if empty
+    Cool,
+}
+
+/// Function to charge extra gas per opcode based on cool cheatcode
+fn add_gas_from_cool_cheatcode<DB: DatabaseExt>(
+    state: &mut Cheatcodes,
+    interpreter: &mut Interpreter,
+    data: &mut EVMData<'_, DB>,
+) -> InstructionResult {
```

Comment:
This result is not used

Reply:
looks like I added this because I copied the use of `try_or_continue!(interpreter.stack().peek(0))` which returns `InstructionResult::Continue` on Err, but then step_end has an eval that is InstructionResult so wasn't sure if that is supposed to be returned instead? Otherwise I'd just inline the function back to make it work

Reply:
Ah that's fine then